### PR TITLE
Transcribe fix

### PR DESCRIPTION
**Pull request by II245**

### CODE REVIEW

File: datasets/youtube.py [hunk 1]
```diff
@@ -3,30 +3,40 @@
 import argparse
 import glob
 
+def clean_up_transcript_ref(ref):
```
Comment (vadimkantorov) [inline datasets/youtube.py:6]: why is this needed? isn't this handled by preprocessor? are there other offending punct marks? shouldn't this be replaced by space?

File: scripts/youtube_dataset_creation.sh [hunk 2]
```diff
@@ -6,35 +6,35 @@ THREAD=$2
...
 
-TRANSCRIPT_PREPROC='--split-by-parts 3 --skip-files-longer-than-hours 4 --skip-transcript-after-seconds 1800 '
-TRANSCRIBE='--mono --batch-time-padding-multiple 1 --align --skip-processed --max-segment-duration 4.0 --transcribe-first-n-sec 3600'
-SUBSET='--num-speakers 1 --gap 0.05- --cer 0.0-0.25 --duration 0.5-8.0'
+TRANSCRIPT_PREPROC='--split-by-parts 3 --skip-files-longer-than-hours 4 --skip-transcript-after-seconds 3600'
+TRANSCRIBE='--mono --join-transcript --batch-time-padding-multiple 1 --align --skip-processed --max-segment-duration 4.0 --transcribe-first-n-sec 3600 --text-pipelines dataset'
+SUBSET='--num-speakers 1 --gap 0.05- --cer 0.0-0.35 --duration 0.5-8.0'
 CUT="--dilate 0.025 --sample-rate $SAMPLE_RATE --mono --strip-prefix data/ --add-sub-paths --strip"
-TRAIN_TEST_SPLIT='--microval-duration-in-hours 0 --val-duration-in-hours 0 --test-duration-in-hours 0 --old-microval-path cut_microval_10h.json'
+TRAIN_TEST_SPLIT='--microval-duration-in-hours 0 --val-duration-in-hours 0 --test-duration-in-hours 0 --seed 2021'
```
Comment (vadimkantorov) [inline scripts/youtube_dataset_creation.sh]: please move seed into a variable

File: scripts/youtube_dataset_creation.sh [hunk 3]
```diff
@@ -6,35 +6,35 @@ THREAD=$2
...
 #bash datasets/youtube.sh LIST "$YOUTUBE" > $DATASET_ROOT/youtube.txt
 #bash datasets/youtube.sh LIST 'https://www.youtube.com/channel/UCb9_Bhv37NXN1m8Bmrm9x9w'
 #head -n $SAMPLE $DATASET_ROOT/youtube.txt > $DATASET_AUDIO/audio.txt # list of links like http://youtu.be/e8bkFQD3ZhA
 #bash datasets/youtube.sh RETR $DATASET_AUDIO/audio.txt $DATASET_AUDIO
 
-python3 datasets/youtube.py -i "$DATASET_AUDIO_MASK" -o $DATASET_AUDIO.json $TRANSCRIPT_PREPROC
+#python3 datasets/youtube.py -i "$DATASET_AUDIO_MASK" -o $DATASET_AUDIO.json $TRANSCRIPT_PREPROC
 
-#python3 transcribe.py --checkpoint $CHECKPOINT -i $DATASET_AUDIO.json -o $DATASET_TRANSCRIBE $TRANSCRIBE
+python3 transcribe.py --checkpoint $CHECKPOINT -i $DATASET_AUDIO$THREAD.json -o $DATASET_TRANSCRIBE $TRANSCRIBE --output-json
```
Comment (vadimkantorov) [inline scripts/youtube_dataset_creation.sh]: what semantics does THERAD variable have?

File: scripts/youtube_dataset_creation.sh [hunk 4]
```diff
@@ -6,35 +6,35 @@ THREAD=$2
 export CUDA_VISIBLE_DEVICES=$CUDA_DEVICE
 
 YOUTUBE='https://www.youtube.com/channel/UCb9_Bhv37NXN1m8Bmrm9x9w'
-CHECKPOINT=best_checkpoints/JasperNetBig_NovoGrad_lr1e-4_wd1e-3_bs1024____long-train_bs1024_step6_430k_checkpoint_epoch247_iter0446576.pt
```
Comment (vadimkantorov) [inline scripts/youtube_dataset_creation.sh:9]: given that `dataset_tools` is in better shape now, please move this this file there

File: transcribe.py [hunk 5]
```diff
@@ -131,9 +132,12 @@ def main(args, ext_json = ['.json', '.json.gz']):
 		try:
 			tic = time.time()
 			y, ylen = y.to(args.device), ylen.to(args.device)
+			print('Input:', audio_name)
+			print('Xlen:', x.shape[-1] / args.sample_rate)
```
Comment (vadimkantorov) [inline transcribe.py]: what units does `xlen / args.sample_rate` have? Seconds? spectrogram output? Does `--frontend-in-model` option affect this? Please figure this out and add appropriate units to the print message

File: transcribe.py [hunk 6]
```diff
@@ -131,9 +132,12 @@ def main(args, ext_json = ['.json', '.json.gz']):
 		try:
 			tic = time.time()
 			y, ylen = y.to(args.device), ylen.to(args.device)
+			print('Input:', audio_name)
```
Comment (vadimkantorov) [inline transcribe.py]: Please add full audio_path as well, it's more useful for debugging

File: transcripts.py [hunk 7]
```diff
@@ -266,7 +268,12 @@ def join_transcript(transcript: Transcript, join_channels: bool = False):
 		ref = speaker_phrase_separator.join(t['ref'].strip() for t in transcript)
 		speaker = [t['speaker'] for t in transcript]
 		speaker_name = ','.join(collect_speaker_names(transcript))
-		duration = audio.compute_duration(transcript[0]['audio_path'])
+
+		if duration_from_transcripts:
+			duration = max(t['end'] for t in transcript)
```
Comment (vadimkantorov) [inline transcripts.py]: please use `summary(transcript)['end']` instead

File: datasets/youtube.py (hunk 1 above)
Comment (II245) [reply datasets/youtube.py:6]: We use the replica separator as a semicolon.
If a semicolon is encountered in the reference transcription, the asert in encode_transcript func stops working.
assert len(ref_by_speaker) == len(t['speaker'])

File: scripts/youtube_dataset_creation.sh (hunk 2 above)
Comment (II245) [reply scripts/youtube_dataset_creation.sh]: Remove seed. Left as it was before.

File: scripts/youtube_dataset_creation.sh (hunk 3 above)
Comment (II245) [reply scripts/youtube_dataset_creation.sh]: Thread it is number of audio.json subfile. audio0.json represent part of whole audio.json.

File: datasets/youtube.py (hunk 1 above)
Comment (vadimkantorov) [reply datasets/youtube.py:6]: please put semicolon in a properly named variable and add a comment

File: transcribe.py (hunk 6 above)
Comment (II245) [reply transcribe.py]: done

Comment (vadimkantorov) [reply transcribe.py]: maybe the fix is incoming, can't see it yet

File: scripts/youtube_dataset_creation.sh (hunk 4 above)
Comment (II245) [reply scripts/youtube_dataset_creation.sh:9]: done

File: transcribe.py (hunk 5 above)
Comment (II245) [reply transcribe.py]: done

File: configs/ru_text_config.json [hunk 8]
```diff
@@ -43,6 +52,14 @@
...
                 "normalize_text" : true,
                 "allowed_chars": "абвгдежзийклмнопрстуфхцчшщъыьэюя "},
+    "dataset": {"drop_space_at_borders" : false,
+                "to_lower_case" : true,
+                "collapse_char_series" : false,
+                "drop_substrings" : ["|", "*"],
+                "replace_chars" :  [],
+                "repeat_character" : "2",
+                "normalize_text" : false,
+                "allowed_chars": "абвгдеёжзийклмнопрстуфхцчшщъыьэюя2 "},
```
Comment (Squire-tomsk) [inline configs/ru_text_config.json]: Is it ok to allow "2" symbol after postprocessing?

File: transcribe.py [hunk 9]
```diff
@@ -131,9 +132,11 @@ def main(args, ext_json = ['.json', '.json.gz']):
 		try:
 			tic = time.time()
 			y, ylen = y.to(args.device), ylen.to(args.device)
+			print('Input path:', meta[0]['audio_path'])
```
Comment (vadimkantorov) [inline transcribe.py:134]: please move these prints before the `y` moves to device to guard against bad OOM

File: configs/ru_text_config.json (hunk 8 above)
Comment (II245) [reply configs/ru_text_config.json]: remove 2

File: transcribe.py (hunk 9 above)
Comment (II245) [reply transcribe.py:134]: done

File: transcripts.py (hunk 7 above)
Comment (II245) [reply transcripts.py]: done

File: datasets/youtube.py (hunk 1 above)
Comment (II245) [reply datasets/youtube.py:6]: done
